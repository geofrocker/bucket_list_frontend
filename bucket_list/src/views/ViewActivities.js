import React, {Component} from 'react';
import axios from 'axios';
import Header from '../components/Header';
import Sidebar from '../components/Sidebar';
import  {Redirect, Link} from 'react-router-dom'

class ViewActivities extends Component{

    constructor(props){
        super(props);
        let bucket_id = this.props.match.params.bucket_id;
        this.state = {description: '', date_created: '', updated: '', isAuthenticated:false, data:[], bucket_id:bucket_id};
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(){
        return (
            <Redirect to = {'/bucketlists/' + this.state.bucket_id + '/items/create/'} />
        )}
    
    componentDidMount() {
        axios({
            url: 'http://127.0.0.1:5000/api/v1/bucketlists/' + this.state.bucket_id + '/items',
            method: "GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        })
            .then((response)=>{
                this.setState({data:response.data.activities, isAuthorized: true});
                console.log(JSON.stringify(response))

            })
            .catch((xhr) =>{
                console.log(JSON.stringify(xhr))
            });
    }

    render(){
        if (!this.state.isAuthorized){
            return(
                <div>
                    <article className="content item-editor-page">
                        <div className="card card-block">
                            <p>Unauthorized! Please log in</p>
                        </div>
                    </article>
                </div>
            )
        }
        return(
            <div>
                <Header/>
                <Sidebar/>
                <article className="content responsive-tables-page">
                    <div className="title-block">
                        <h1 className="title">Activities</h1>
                    </div>

                    <section className="section">
                        <div className="row">
                            <div className="col-md-12">
                                <div className="card">
                                    <div className="card-block">
                                        <div className="card-title-block">
                                            <button type="button" onClick={this.handleClick} className="btn btn-primary-outline">Add New</button>
                                        </div>

                                    </div>
                                    <section className="example">
                                        <div className="table-flip-scroll">
                                            <table className="table table-striped table-bordered table-hover flip-content">
                                                <thead className="flip-header">
                                                <tr>
                                                    <th>Description</th>
                                                    <th>Date Created</th>
                                                    <th>Update</th>
                                                    <th>Action</th>
                                                </tr>
                                                </thead>
                                                <tbody>
                                                {this.state.data.map(function(item, key){
                                                    return (
                                                        <tr key={key}>
                                                            <td>{item.description}</td>
                                                            <td>{item.created}</td>
                                                            <td className="center" >
                                                                <Link to={'/bucketlists/' + item.bucket_id + '/items/update/' + item.activity_id}><td>Update</td></Link>
                                                            </td>
                                                            <td className="center">
                                                                <button type="button"  className="btn btn-danger-outline btn-sm">Delete</button>

                                                            </td>
                                                        </tr>
                                                    )
                                                })}
                                                </tbody>
                                            </table>
                                        </div>
                                    </section>
                                </div>
                            </div>
                        </div>
                    </section>
                </article>
            </div>
        )
    }
}
export default ViewActivities