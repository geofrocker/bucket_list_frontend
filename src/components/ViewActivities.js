import React, {Component} from 'react';
import axios from 'axios';
import Header from './Header';
import Sidebar from './Sidebar';
import  {Redirect, Link} from 'react-router-dom'


class ViewActivities extends Component{
    /**
     * This component is used to create activities and items.
     * It handles the following methods:
     *  handleClick --> used to redirect to the create items url
     *  render --> Used to return jsx
     */

    constructor(props){
        super(props);
        let bucket_id = this.props.match.params.bucket_id;
        this.state = {description: '', date_created: '', updated: '', isAuthenticated:false, data:[], bucket_id:bucket_id, login_redirect:false};
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(){
        return (
            <Redirect to = {'/bucketlists/' + this.state.bucket_id + '/items/create/'} />
        )}
    
    componentDidMount() {
        axios({
            url: 'http://127.0.0.1:5000/api/v1/bucketlists/' + this.state.bucket_id + '/items',
            method: "GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        })
            .then((response)=>{
                this.setState({data:response.data.activities, isAuthorized: true});
                console.log(JSON.stringify(response))

            })
            .catch((xhr) =>{
                this.setState({login_redirect:true})
            });
    }

    render(){
        if(this.state.login_redirect){
            return(
                <Redirect to={'/login/'}/>
            )
        }

        else if (!this.state.isAuthorized){
            return(
                <div>
                    <article className="content item-editor-page">

                    </article>
                </div>
            )
        }
        return(
            <div>
                <Header/>
                <Sidebar/>
                <article className="content responsive-tables-page">
                    <div className="title-block">
                        <h1 className="title">Activities</h1>
                    </div>

                    <section className="section">
                        <div className="row">
                            <div className="col-md-12">
                                <div className="card">
                                    <div className="card-block">
                                        <div className="card-title-block">
                                            <button type="button" className="btn btn-primary-outline">
                                                <Link to={"/bucketlists/" + this.state.bucket_id + "/items/create/"}>Add New</Link>
                                            </button>
                                        </div>

                                    </div>
                                    <section className="example">
                                        <div className="table-flip-scroll">
                                            <table id="activities_table" className="table table-striped table-bordered table-hover flip-content">
                                                <thead className="flip-header">
                                                <tr>
                                                    <th>Description</th>
                                                    <th>Date Created</th>
                                                    <th>Update</th>
                                                    <th>Action</th>
                                                </tr>
                                                </thead>
                                                <tbody>
                                                {this.state.data.map(function(item, key){
                                                    return (
                                                        <tr key={key}>
                                                            <td>{item.description}</td>
                                                            <td>{item.created}</td>
                                                            <td className="center" >
                                                                <Link to={'/bucketlists/' + item.bucket_id + '/items/update/' + item.activity_id}><td>Update</td></Link>
                                                            </td>
                                                            <td className="center" >
                                                                <button type="button" className="btn btn-danger-outline btn-sm">
                                                                    <Link to={'/bucketlists/' + item.bucket_id + '/items/delete/' + item.activity_id}><td>Delete</td></Link>
                                                                </button>
                                                            </td>
                                                        </tr>
                                                    )
                                                })}
                                                </tbody>
                                            </table>
                                        </div>
                                    </section>
                                </div>
                            </div>
                        </div>
                    </section>
                </article>
            </div>
        )
    }
}
export default ViewActivities