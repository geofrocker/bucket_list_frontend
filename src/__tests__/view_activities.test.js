import React from 'react';
import ViewActivities from '../views/ViewActivities';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {MemoryRouter} from 'react-router-dom';

describe('<ViewActivities />', ()=>{
    beforeEach(() => {
        global.localStorage = {
            token: 'someToken',
            getItem: function () {
                return 'someToken'
            }
        };

    });

    it('contains bucket table', function () {
        const props = {
            match:{
                params: {bucket_id:1}
            }
        };
        const wrapper = mount(
            <MemoryRouter>
                <ViewActivities {...props}/>
            </MemoryRouter>);

        wrapper.setState({data:[{ description:"test description"}], isAuthorized:true});

        setTimeout(() => {
            expect(wrapper.find("#activities_table").length === 1);

        }, 60);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});

