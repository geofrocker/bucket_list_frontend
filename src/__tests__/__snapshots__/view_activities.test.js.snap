// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewActivities /> contains bucket table 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "vv6ias",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "vv6ias",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ViewActivities
      match={
        Object {
          "params": Object {
            "bucket_id": 1,
          },
        }
      }
    >
      <div>
        <article
          className="content item-editor-page"
        >
          <div
            className="card card-block"
          >
            <p>
              Unauthorized! Please 
              <Link
                replace={false}
                to="/login/"
              >
                <a
                  href="/login/"
                  onClick={[Function]}
                >
                  Login
                </a>
              </Link>
            </p>
          </div>
        </article>
      </div>
    </ViewActivities>
  </Router>
</MemoryRouter>
`;
